server/api/plugins.js

```diff
@@ -2,18 +2,91 @@ import {parseMarkdown} from '@nuxtjs/mdc/runtime'
 import url from "node:url";
 import {camelToKebabCase} from "~/utils/url.js";
 
-function toNuxtContent(parsedMarkdown) {
+function toNuxtContent(parsedMarkdown, type) {
     return {
         body: {
             toc: parsedMarkdown.toc,
             ...parsedMarkdown.body
         },
+        pluginType: type,
         description: parsedMarkdown.data.description,
         title: parsedMarkdown.data.title,
         icon: `data:image/svg+xml;base64,${parsedMarkdown.data.icon}` ?? null
     };
 }
 
+function toNuxtBlocks(data, type) {
+    const { $schema, required, title, description, $examples, $deprecated, $beta, $metrics, ...pageData } = data;
+
+    if ($examples) {
+        pageData.examples = $examples;
+    }
+
+    if ($metrics) {
+        pageData.metrics = $metrics;
+    }
+    return {
+        body: {
+            children: pageData,
+            toc: {
+                links: navTocData(pageData)
+            },
+        },
+        pluginType: type,
+        description,
+        deprecated: $deprecated,
+        title,
+    };
+}
+
+const generateNavTocChildren = (properties) => {
+    const children = [];
+
+    for (const key in properties) {
+        if (properties.hasOwnProperty(key)) {
+            children.push({
+                id: key,
+                depth: 3,
+                text: properties[key].name ?  properties[key].name : key,
+                required: properties[key].$required || false
+            })
+        }
+    }
+
+    children.sort((a, b) => b.required - a.required);
```

Comment:
perform your sort on required field before your populating for loop so that you don't need to push the required property to children elements. It will allow you to remove the below map and just return children